### PR TITLE
Defect/de3819 keep data around

### PR DESCRIPTION
**Pull request by JoeKer1**

### CODE REVIEW

File: src/app/components/create-group/create-group-data.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                // { provider: ServiceName, useClass: FakeServiceClass },
+                // { provider: ServiceName, useFactory: fakeServiceFactory, deps: [] },
+            ]
+        });
+    });
+
+    // you can also wrap inject() with async() for asynchronous tasks
+    // it('...', async(inject([...], (...) => {}));
+
+    fit('should create the create group service',
```
Comment (vchtelmakh) [inline src/app/components/create-group/create-group-data.service.spec.ts]: Unfit tests.

File: src/app/components/create-group/create-group-data.service.ts [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+            return this.lookupService.getCategories()
+            .do((cats: Category[]) => {
+                this.categories = cats;
+                this.pageOneInitialized = true;
+            });
+        } else {
+            return Observable.of(this.categories);
+        }
+    }
+
```
Comment (vchtelmakh) [inline src/app/components/create-group/create-group-data.service.ts]: Shouldn't categories come from a route resolve if they are always required?

Comment (vchtelmakh) [inline src/app/components/create-group/create-group-data.service.ts]: Should we consider preloading categories in the component that requires them, in a route guard?

File: src/app/components/create-group/create-group-data.service.spec.ts (hunk 1 above)
Comment (JoeKer1) [inline src/app/components/create-group/create-group-data.service.spec.ts]: dang it i knew i forgot something

File: src/app/components/create-group/create-group-data.service.ts (hunk 2 above)
Comment (JoeKer1) [inline src/app/components/create-group/create-group-data.service.ts]: I suppose we could move this logic into a guard but then we'd have to load it into the service on submit of the form and then the guard would have to get the data from the service. Shouldn't be a big deal if that's what we wanna do.

Comment (vchtelmakh) [inline src/app/components/create-group/create-group-data.service.ts]: " to load it into the service on submit of the form and then the guard would have to get the data from the service or the backend depending on if it has been initialized" We could pass the cats to whatever service method uses them.

File: src/app/components/create-group/page-1/create-group-page-1.component.ts [hunk 3]
```diff
@@ -1,31 +1,60 @@
...
+        }
+    }
+
+    public onSubmit(form) {
+        this.isSubmitted = true;
+        if (form.valid) {
+            // Do Something
+        } else {
+            // Do something else
+        }
```
Comment (vchtelmakh) [inline src/app/components/create-group/page-1/create-group-page-1.component.ts]: Do we need to do something here, or are these placeholders for when we get more pages?

Comment (JoeKer1) [inline src/app/components/create-group/page-1/create-group-page-1.component.ts]: Placeholders until we get more pages

File: config/webpack.test.js [hunk 4]
```diff
@@ -51,6 +52,14 @@ module.exports = function(options) {
...
         systemvars: true
       }),
+
+      new webpack.ContextReplacementPlugin(
+         // The (\\|\/) piece accounts for path separators in *nix and Windows
+         /angular(\\|\/)core(\\|\/)@angular/,
+         helpers.root('./src'), // location of your src
+         { }
+       ),
+
```
Comment (Doug-Shannon) [inline config/webpack.test.js]: Nice!

File: src/app/components/create-group/page-1/create-group-page-1.component.ts [hunk 5]
```diff
@@ -1,31 +1,60 @@
...
+
+    constructor(private state: StateService,
+                private createGroupService: CreateGroupService) { }
 
     ngOnInit() {
+        this.state.setLoading(true);
         this.state.setPageHeader('start a group', '/create-group');
-        this.initializeCategories();
+        this.groupCategoryForm = new FormGroup({});
+        this.createGroupService.initializePageOne()
```
Comment (Doug-Shannon) [inline src/app/components/create-group/page-1/create-group-page-1.component.ts]: if you want me to be nitpicky, I don't like trailing dot notation.

Comment (JoeKer1) [inline src/app/components/create-group/page-1/create-group-page-1.component.ts]: I do want you to be nitpicky. I don't like it either but it helps me keep track of like finally().subscribe() etc. 
Do you prefer: 
`this.createGroupService.initializePageOne().finally(
() => {
  this.state.setLoading(false);
}).subscribe(cats => {
  this.initializeCategories(cats);
});`

**Pull request merged**